recipes/lightgbm/all/conanfile.py

```diff
@@ -1,82 +1,115 @@
     homepage = "https://github.com/microsoft/LightGBM"
-    license = "MIT"
+    topics = ("machine-learning", "boosting")
 
+    package_type = "library"
     settings = "os", "arch", "compiler", "build_type"
-    exports_sources = ["CMakeLists.txt", "patches/**"]
-    generators = "cmake", "cmake_find_package"
     options = {
         "shared": [True, False],
         "fPIC": [True, False],
-        "with_openmp": [True, False]
+        "with_openmp": [True, False],
     }
     default_options = {
         "shared": False,
         "fPIC": True,
-        "with_openmp": True
+        "with_openmp": True,
     }
 
-    @property
-    def _source_subfolder(self):
-        return "source_subfolder"
-
     def config_options(self):
         if self.settings.os == "Windows":
             del self.options.fPIC
 
     def configure(self):
         if self.options.shared:
-            del self.options.fPIC
+            self.options.rm_safe("fPIC")
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
 
     def requirements(self):
         self.requires("eigen/3.4.0")
-        self.requires("fast_double_parser/0.6.0")
-        self.requires("fmt/9.0.0")
-        if self.options.with_openmp and self.settings.compiler in ("clang", "apple-clang"):
-            self.requires("llvm-openmp/11.1.0")
+        self.requires("fast_double_parser/0.7.0", transitive_headers=True)
+        self.requires("fmt/10.0.0", transitive_headers=True)
+        # FIXME: enable when llvm-openmp has been migrated to Conan v2
+        # if self.options.with_openmp and self.settings.compiler in ("clang", "apple-clang"):
+        #     self.requires("llvm-openmp/12.0.1")
 
     def validate(self):
         if self.settings.compiler.get_safe("cppstd"):
-            tools.check_min_cppstd(self, 11)
+            check_min_cppstd(self, 11)
 
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version],
-                  destination=self._source_subfolder, strip_root=True)
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.cache_variables["BUILD_STATIC_LIB"] = not self.options.shared
+        tc.cache_variables["USE_DEBUG"] = self.settings.build_type in ["Debug", "RelWithDebInfo"]
+        tc.cache_variables["USE_OPENMP"] = self.options.with_openmp
+        tc.cache_variables["BUILD_CLI"] = False
+        if self.settings.os == "Macos":
+            tc.cache_variables["APPLE_OUTPUT_DYLIB"] = True
+        tc.generate()
+        tc = CMakeDeps(self)
+        tc.generate()
 
     def _patch_sources(self):
-        for patch in self.conan_data.get("patches", {}).get(self.version, []):
-            tools.patch(**patch)
+        cmakelists_path = os.path.join(self.source_folder, "CMakeLists.txt")
+        # Fix OpenMP detection for Clang
+        replace_in_file(self, cmakelists_path, "AppleClang", "Clang|AppleClang")
```

Comment:
I would say it may worth a patch to the upstream, as it's small and we always need to apply it.